src/EmbedIO/Sessions/SessionProxy.cs

```diff
@@ -83,6 +83,8 @@ public DateTime LastActivity
         /// <inheritdoc/>
         public void Delete()
         {
+            EnsureSessionExists();
```

Comment:
You are creating a session in order to delete it.

Reply:
This was a bug, the session was not restored from the manager before delete, so the following conditional returns immediately, Maybe using ensure session exists with an optional argument for not creation.